modules/core/src/main/java/org/apache/ignite/internal/commandline/cache/CheckIndexInlineSizes.java

```diff
@@ -0,0 +1,176 @@
+
+/**
+ * Command for check secondary indexes inline size on the different nodes.
+ */
+public class CheckIndexInlineSizes implements Command<Void> {
+    /** Warn message format. */
+    public static final String NOT_ALL_NODES_SUPPORT_FEATURE_WARN_MSG_FMT =
+        "Indexes inline size were not checked on all nodes because the few nodes aren't support this feature. Skipped nodes: %s";
+
+    /** Success message. */
+    public static final String INDEXES_INLINE_SIZE_ARE_SAME =
+        "All secondary indexes have the same effective inline size on all cluster nodes.";
+
+    /** Predicate to filter server nodes. */
+    private static final Predicate<GridClientNode> SRV_NODES = node ->
+        Objects.equals(node.attribute(ATTR_CLIENT_MODE), false) && !"true".equals(node.attribute(ATTR_DAEMON));
+
+    /** {@inheritDoc} */
+    @Override public Object execute(GridClientConfiguration clientCfg, Logger log) throws Exception {
+        try (GridClient client = Command.startClient(clientCfg)) {
+            Set<GridClientNode> serverNodes = client.compute().nodes().stream()
+                .filter(SRV_NODES)
+                .collect(toSet());
+
+            Set<GridClientNode> supportedServerNodes = serverNodes.stream()
+                .filter(n -> checkIndexInlineSizesSupported(n))
+                .collect(toSet());
+
+            Collection<UUID> serverNodeIds = F.transform(serverNodes, GridClientNode::nodeId);
+            Collection<UUID> supportedServerNodeIds = F.transform(supportedServerNodes, GridClientNode::nodeId);
+
+            CheckIndexInlineSizesResult res = client.compute().projection(supportedServerNodes).execute(
+                CheckIndexInlineSizesTask.class.getName(),
+                new VisorTaskArgument<>(supportedServerNodeIds, false)
+            );
+
+            Set<UUID> unsupportedNodes = serverNodeIds.stream()
+                .filter(n -> !supportedServerNodeIds.contains(n))
+                .collect(toSet());
+
+            analizeResults(log, unsupportedNodes, res);
+        }
+
+        return null;
+    }
+
+    /** */
+    private void analizeResults(
+        Logger log,
+        Set<UUID> unsupportedNodes,
+        CheckIndexInlineSizesResult res
+    ) {
+        if (!F.isEmpty(unsupportedNodes))
+            log.info(String.format(NOT_ALL_NODES_SUPPORT_FEATURE_WARN_MSG_FMT, unsupportedNodes));
+
+        Map<String, Map<Integer, Set<UUID>>> indexToSizeNode = new HashMap<>();
+
+        for (Map.Entry<UUID, Map<String, Integer>> nodeRes : res.inlineSizes().entrySet()) {
+            for (Map.Entry<String, Integer> index : nodeRes.getValue().entrySet()) {
+                Map<Integer, Set<UUID>> sizeToNodes = indexToSizeNode.computeIfAbsent(index.getKey(), x -> new HashMap<>());
+
+                sizeToNodes.computeIfAbsent(index.getValue(), x -> new HashSet<>()).add(nodeRes.getKey());
+            }
+        }
+
+        log.info("Found " + indexToSizeNode.size() + " secondary indexes.");
```

Comment:
1. one more "secondary" word )
2. and what about grid with PK indexes only ? empty result will be printing, isn`t it ?

Reply:
Yes, we check only secondary indexes. primary and affinity key indexes will be skipped during the check.